cfripper/rules/S3CrossAccountTrustRule.py

```diff
@@ -29,6 +29,11 @@ class S3CrossAccountTrustRule(Rule):
     REASON = "{} has forbidden cross-account policy allow with {} for an S3 bucket."
 
     def invoke(self, cfmodel):
+        valid_principals = {
+            *self._config.aws_service_accounts,
+            *self._config.aws_principals,
+            self._config.aws_account_id,
+        }
```

Comment:
This valid principals snippet appears 4 times. Extract it into a property or something